pkgs/tools/security/pinentry/mac.nix

```
@@ -43,8 +43,11 @@ stdenv.mkDerivation rec {
   configureFlags = [ "--enable-maintainer-mode" "--disable-ncurses" ];
 
   installPhase = ''
-    mkdir -p $out/Applications
+    mkdir -p $out/Applications $out/bin
     mv macosx/pinentry-mac.app $out/Applications
+
+    # Compatibility with `lib.getExe`
+    ln -s $out/Applications/pinentry-mac.app/Contents/MacOS/pinentry-mac $out/bin/pinentry-mac
```

```suggestion
    makeWrapper $out/Applications/pinentry-mac.app/Contents/MacOS/pinentry-mac $out/bin/pinentry-mac
```

with `makeWrapper` or `makeBinaryWrapper` added to `nativeBuildInputs`.